crates/sui-types/src/temporary_store.rs

```diff
@@ -670,7 +670,19 @@ impl<S> TemporaryStore<S> {
         }
         let cost_summary = gas_status.summary(result.is_ok());
         let gas_used = cost_summary.gas_used();
-        let gas_rebate = (cost_summary.storage_rebate as f64 * STORAGE_REBATE_RATE).round() as u64;
+        // value as computed before
+        let gas_rebate1 = (cost_summary.storage_rebate as f64 * 0.99).round() as u64;
+        // computation with u64 arithmetic
+        // multiply by STORAGE_REBATE_RATE first value (numerator)
+        let mut gas_rebate =
+            (cost_summary.storage_rebate as u128 * STORAGE_REBATE_RATE.0 as u128) as u64;
+        // divide by STORAGE_REBATE_RATE second value and round
+        gas_rebate = ((gas_rebate as u128 + (STORAGE_REBATE_RATE.1 as u128 / 2))
```

Comment:
Why not just either rounding up or rounding down? I don't have strong preference but this line feels a little confusing.

Reply:
we can do both and break the previous behavior, which is fine by me but I would love for us to think about it.
Bottom line the old code has a `round()` call, some test rely on it and I am not sure we what we want to do.
You'll notice the next line (commented) is asking exactly that question

Reply:
I will add that my vote is for truncation, the simplest and most clear way to do it. I believe also more reliable as the value rounding down would less likely create issue with gas

Reply:
moved to truncation

Reply:
and back to rounding which makes things working and easy